src/main/java/de/blau/android/bookmarks/BookmarkIO.java

```diff
@@ -0,0 +1,87 @@
+package de.blau.android.bookmarks;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.blau.android.osm.ViewBox;
+import de.blau.android.util.SavingHelper;
+
+/**
+ * Handles Bookmark reading/writing operations
+ */
+public class BookmarkIO {
+    private BookmarksStorage currentBookmarkStorage;
+    private List<BookmarksStorage> bookmarksStorage;
+    private final String fileName = "bookmarks.ser";
+    SavingHelper<ArrayList<BookmarksStorage>> savingHelper;
+
+    /**
+     * BookmarkIO constructor
+     */
+    public BookmarkIO() {
+        this.currentBookmarkStorage = new BookmarksStorage();
+        this.bookmarksStorage = new ArrayList<>();
+        this.savingHelper = new SavingHelper<>();
+    }
+
+    /**
+     * Adds the text and viewbox to a list
+     *
+     * @param comment Bookmark name/comment
+     * @param viewbox Map Viewbox
+     */
+    public void addDatatolist(@NonNull String comment, @NonNull ViewBox viewbox) {
+        currentBookmarkStorage.set(comment, viewbox);
+        bookmarksStorage.add(currentBookmarkStorage);
+    }
+
+    /**
+     * (Over)Writes the bookmark file to storage.
+     *
+     * @param context          the Android Context
+     * @param bookmarksStorage Arraylist containing BookmarksStorage objects
+     * @return true if successful
+     */
+    public boolean writeList(@NonNull Context context, @NonNull ArrayList<BookmarksStorage> bookmarksStorage) {
+        return savingHelper.save(context, fileName, bookmarksStorage, true);
+    }
+
+    /**
+     * Reads the bookmark file from storage
+     *
+     * @param context the Android context
+     * @return Arraylist containg saved bookmarks
+     */
+
+    @NonNull
+    public List<BookmarksStorage> readList(@NonNull Context context) {
+        List<BookmarksStorage> savedList = savingHelper.load(context, fileName, true);
+        if (savedList == null) {
+            // returns an empty list instead of nu
+            return this.bookmarksStorage;
+        }
+        return savedList;
+    }
+
+    /**
+     * Utility for a saving bookmark
+     *
+     * @param context  the Android context
+     * @param comments Bookmark name/comment
+     * @param viewBox  map viewbox
+     */
+    public void writer(@NonNull Context context, @NonNull String comments, @NonNull ViewBox viewBox) {
+        this.bookmarksStorage = readList(context);
+        addDatatolist(comments, viewBox);
+        writeList(context, (ArrayList<BookmarksStorage>) this.bookmarksStorage);
```

Comment:
The cast to ArrayList<BookmarksStorage> should be removed as the argument needs to be an ArrayList for SavingHelper in any case (my bad that I didn't spot this).